pkg/bwagreement/test/server_test.go

```diff
@@ -8,35 +8,113 @@ import (
 	"crypto/ecdsa"
 	"testing"
 
+	//"github.com/gogo/protobuf/proto"
 	"github.com/stretchr/testify/assert"
 	"go.uber.org/zap"
 
 	"storj.io/storj/internal/testcontext"
 	"storj.io/storj/internal/testidentity"
+	"storj.io/storj/internal/teststorj"
 	"storj.io/storj/pkg/bwagreement"
 	"storj.io/storj/pkg/pb"
 	"storj.io/storj/satellite"
 	"storj.io/storj/satellite/satellitedb/satellitedbtest"
 )
 
-func TestBandwidthAgreements(t *testing.T) {
+func TestSameSerialNumberBandwidthAgreements(t *testing.T) {
 	satellitedbtest.Run(t, func(t *testing.T, db satellite.DB) {
 		ctx := testcontext.New(t)
 		defer ctx.Cleanup()
 
+		/* More than one storagenode can submit bwagreements with the same serial number.
+		   Uplink would like to download a file from 2 storagenodes.
+		   Uplink requests a PayerBandwidthAllocation from the satellite. One serial number for all storagenodes.
+		   Uplink signes 2 RenterBandwidthAllocation for bothboth storagenode. */
```

Comment:
Typo: "bothboth".
Also "storage node" reads easier than "storagenode' in comments.